server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -885,6 +885,10 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
     @DB
     protected void updateSite2SiteVpnConnectionState(final List<DomainRouterVO> routers) {
         for (final DomainRouterVO router : routers) {
+            if (router.getRole() != Role.VIRTUAL_ROUTER) {
```

Comment:
Could it cause regressions for anything others: `LB, INTERNAL_LB_VM, NETSCALER_VM` cc @weizhouapache @DaanHoogland

Reply:
@rhtyd as far as I know, site2site vpn can only be applied on vpc vrs, so this should be ok.

Reply:
@weizhouapache @ravening cc @DaanHoogland - the only type I'm not sure about is Netscalars. @PaulAngus can you comment if netscalar supports site-to-site VPN?

Reply:
Do we have netscaler users present and active?

Reply:
We may that's why I've pinged @PaulAngus - advice?

Reply:
Ping @andrijapanicsb @PaulAngus

Reply:
not sure I can advise here... I say we go forward, as NetScaller should definitively NOT be marked as "VIRTUAL_ROUTER"

Reply:
@ravening can you change the conditional to: `router.getRole() == INTERNAL_LB_VM`

Reply:
I have made necessary changes